packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -151,6 +146,99 @@ async function reconciliationReportForCollections() {
   };
 }
 
+/**
+ * Compare the granule holdings in CMR with Cumulus
+ *
+ * @param {string} collectionId - the collection which has the granules to be reconciled
+ * @returns {Promise<Object>} an object with the okGranulesInDb, okGranulesInCmr, onlyInCumulus,
+ * onlyInCmr
+ */
+async function reconciliationReportForGranules(collectionId) {
+  // compare granule holdings:
+  //   Get CMR granules list (by PROVIDER, short_name, version, sort_key: ['granule_ur'])
+  //   Get CUMULUS granules list (by collectionId order by granuleId)
+  //   Report granules only in CMR
+  //   Report granules only in CUMULUS
+  const { name, version } = deconstructCollectionId(collectionId);
+  const cmrGranulesIterator = new CMRSearchConceptQueue(
+    process.env.cmr_provider, process.env.cmr_client_id, 'granules',
+    { short_name: name, version: version, sort_key: ['granule_ur'] }, 'umm_json'
+  );
+
+  const dbGranulesIterator = new Granule().getGranulesForCollection(collectionId);
+
+  const okGranulesInDb = [];
+  const okGranulesInCmr = [];
+  const granulesOnlyInDb = [];
+  const granulesOnlyInCmr = [];
+
+  let [nextDbItem, nextCmrItem] = await Promise.all([dbGranulesIterator.peek(), cmrGranulesIterator.peek()]); // eslint-disable-line max-len
+
+  while (nextDbItem && nextCmrItem) {
+    const nextDbGranuleId = nextDbItem.granuleId;
+    const nextCmrGranuleId = nextCmrItem.umm.GranuleUR;
+
+    if (nextDbGranuleId < nextCmrGranuleId) {
```

Comment:
This is not a request for a change, but how do you know that the granule is only in DB and not CMR when `nextDbGranuleId < nextCmrGranuleId`? Is the idea that because you're iterating over both sets, once the IDs are different then you know there is a mismatch between the systems? But I can't quite parse how if `nextDbGranuleId < nextCmrGranuleId`, it means that the granule is only in the DB

Reply:
@kkelly51 @markdboyd The granuleIds are sorted in both DB and CMR queues.  The granules from DB are sorted by the granuleId bc the query is against the Global Secondary Index.  The granules from CMR are sorted by the sort_key granule_ur. 
If the DB queue has `granule1, granule3, granule5` and CMR queue has `granule2, granule3`, then `nextDbGranuleId < nextCmrGranuleId` is the string comparison, and `granule1` is only in DB.